3_ecosystem/3_3_date_time/src/main.rs

```diff
@@ -1,21 +1,43 @@
+use chrono::prelude::*;
+
+
 fn main()
 {
   println!( "Implement me!" );
 }
 
 const NOW : &str = "2019-06-26";
 
-struct User;
+struct User
+{
+  birthdate : Date< Utc >,
+}
 
 impl User
 {
-  fn with_birthdate( year : i32, month : u32, day : u32 ) -> Self { unimplemented!() }
+  fn with_birthdate( year : i32, month : u32, day : u32 ) -> Self
+  {
+    User{ birthdate : Utc.ymd( year, month, day ) }
```

Comment:
It seems that `Utc.ymd` can panic. Does it?

Please, update realization. The type should handle wrong input correctly. Maybe, you can create the error type `InvalidDate` or return `Result` with static string.